tar/storage/storage_write.c

```diff
@@ -111,6 +114,20 @@ raisesigs(struct storage_write_internal * S)
 				warnp("raise(SIGUSR2)");
 		}
 	}
+
+	/* Print a message if appropriate. */
+	if (tarsnap_opt_uploadprogressbytes != (uint64_t)(-1)) {
+		if (totalout > lastuploadprogress +
+		    tarsnap_opt_uploadprogressbytes) {
+			lastuploadprogress = totalout;
+			if (tarsnap_opt_humanize_numbers)
+				fprintf(stderr, "Uploaded %s\n",
+				    humansize(totalout));
+			else
+				fprintf(stderr, "Uploaded %" PRIu64 "\n",
+				    totalout);
+		}
+	}
```

Comment:
Hmm, should we do this, or just raise a SIGUSR1 to ourself?

Reply:
SIGUSR1 isn't hugely valuable from a "progress" standpoint:

```
$ tarsnap -c --dry-run ~
tarsnap: Removing leading '/' from member names
adding home/td/texmf/tex/generic/pgf/libraries/shapes/pgflibraryshapes.geometric.code.tex (0 B / 132 kB bytes)
adding home/td/texmf/doc/generic/pgf/pgfmanual.pdf (655 kB / 8.0 MB bytes)
adding home/td/texmf/doc/generic/pgf/images/pgfmanual-mindmap-1.pdf (0 B / 147 kB bytes)
adding home/td/texmf/doc/generic/pgf/text-en/pgfmanual-en-library-external.tex (0 B / 47 kB bytes)
```

Granted, it would accomplish the main goal of reassuring users that something is happening.  But the filenames are essentially arbitrary (and many will be skipped entirely!), and the `0 B`, `655 kB`, `0 B`, `0 B` could be confusing (they doesn't convey a huge amount of "progress"!).  If we wanted to go that route, I think it would be preferable to heavily promote `-v` in the docs instead of changing any code.

I considered trying to print something like
```
adding home/td/texmf/tex/generic/pgf/libraries/shapes/pgflibraryshapes.geometric.code.tex (0 B / 132 kB bytes) [uploaded 8 kB total]
adding home/td/texmf/doc/generic/pgf/pgfmanual.pdf (655 kB / 8.0 MB bytes) [uploaded 14 kB total]
```

but getting the info from `tar/storage/storage_write.c` into `tar/siginfo.c` looked like it would be messy.  I could take another look at that if you want, though.

Reply:
Hmm, take a look at what the latest bsdtar code does.  When I hit ^T on FreeBSD I get 

```
In: 8516 files, 1845319680 bytes; Out: 1845319680 bytes, compression 0%
Current: usr/local/llvm38/bin/llc (13434880/32473040 bytes)
```

which seems like it has more functionality than what I wrote in tarsnap and merged to bsdtar.